backend/dataall/modules/dataset_sharing/services/dataset_alarm_service.py

```diff
@@ -19,9 +19,7 @@ def trigger_table_sharing_failure_alarm(
             target_environment: Environment,
     ):
         log.info('Triggering share failure alarm...')
-        subject = (
-            f'ALARM: DATAALL Table {table.GlueTableName} Sharing Failure Notification'
-        )
+        subject = f'ALARM: DATAALL Table {table.GlueTableName} Sharing Failure Notification'[:100]
```

Comment:
I think we can be smarter here. Let's actually remove the dynamic part from the subject so it remains static and nothing ever needs to be cut. I think people can normally create filters/label by sender to identify it's a data.all email and we can see the glue table name inside the text itself? I would apply this for all alarms.

Reply:
I thought about that too. 
I prefer having the information of the table or bucket name in the Subject itself. 
Think about the scenario when users have a tonne of requests to approve and want to prioritize the important ones ones from the failed (or successful) shares. They will have to open each email or notification individually and read through the information to get their answer. If we just put it in the subject, even if it gets truncated, they will atleast be able to read some of it and decide which ones to prioritize. Hence having the 100 character limit makes more sense to me. 

@dlpzx what do you think?

Reply:
I see the value of having the table name in the subject because of the capability to search by table name (or at least first part of the name). But maybe we can move the dynamic part to the end of the string so that we truncate the table/folder/bucket name and not the message itself. Something like:

`subject = f'ALARM: DATAALL Sharing Failure Notification for Table {table.GlueTableName} '[:100]` what do you think?

Reply:
I agree. Will write it this way in next revision.

Reply:
I think it's a fine compromise to put it at the end..